src/components/tree-select.tsx

```diff
@@ -56,19 +59,23 @@ const TreeSelect = <Item extends BasicItem<Item>>({
 }: Except<DropdownButtonProps, 'children' | 'label'> &
   TreeSelectProps<Item>) => {
   const [activeNodes, setActiveNodes] = useState(defaultActiveNodes);
-  const [openNodes, setOpenNodes] = useState<BasicItem<Item>['id'][]>([]);
+
+  const [openNodes, setOpenNodes] = useState<BasicItem<Item>['id'][]>(
+    Array.from(getSingleChildren(data))
+  );
   const [autocompleteShowDropdown, setAutocompleteShowDropdown] =
     useState(false);
 
   const toggleNode = useCallback(
-    (node: AutocompleteItemType | BasicItem<Item>) => {
-      if (openNodes.includes(node.id)) {
-        setOpenNodes(openNodes.slice(0, openNodes.indexOf(node.id)));
-      } else {
-        setOpenNodes([...openNodes, node.id]);
-      }
-    },
-    [openNodes]
+    (node: AutocompleteItemType | BasicItem<Item>) =>
+      setOpenNodes((openNodes) => {
+        const index = openNodes.indexOf(node.id);
+        if (index === -1) {
+          return [...openNodes, node.id];
+        }
+        return openNodes.filter((id) => id !== node.id);
```

Comment:
```suggestion
        return [...openNodes.slice(0, index), ...openNodes.slice(index + 1)];
```
(related to comment below) Or could use the index since we already have it.

Reply:
Went with the previous suggestion instead of this one